Feature/cp 3621 matomo integration

gilclefes opened the pull request:
## Description

* Matomo Integration
* [Add a link to the Jira ticket.](https://oneacrefund.atlassian.net/browse/CP-3621)
* Integrating Matomo analytics and adding codeowners file


## Checklist
<img width="1486" alt="image" src="https://github.com/user-attachments/assets/5c0c8f72-8826-4c01-a2a5-a280683c15da" />


Please make sure these boxes are checked before submitting your pull request - thanks!
- [x] Followed the PR title naming convention mentioned above.

- [x] Design related bullet points or design document link related to this PR added in the description above. 

- [x] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [x] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Integrated comprehensive Matomo analytics tracking across the application, covering page views, user interactions, form submissions, navigation, and business metrics.
  - Added detailed event tracking to user management, payment hub, transactions, audit trails, system settings, and core UI components.
  - Enabled analytics opt-in/out and tracked language and theme changes in settings.
  - Enhanced performance and business metric tracking for exports, filtering, sorting, pagination, and key workflows.

- **Style**
  - Improved formatting and readability across multiple HTML templates and component files.

- **Chores**
  - Added Matomo dependencies and directives.
  - Updated environment and configuration files for analytics support and OAuth settings.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

gilclefes committed:
feat: Implement Matomo Analytics Integration

- Added Matomo analytics tracking to the web app component for page views and events.
- Integrated MatomoService for tracking user interactions, including login/logout, form submissions, and custom events.
- Created directives for click, form, download, and outbound link tracking.
- Enhanced environment configurations to support Matomo settings.
- Documented analytics implementation details and verification status in ANALYTICS_IMPLEMENTATION_SUMMARY.md.

gilclefes committed:
feat: Update Matomo configuration and release notes

gilclefes committed:
fix: Remove NODE_OPTIONS from build scripts in package.json

gilclefes commented on package.json:
```diff
@@ -43,6 +43,8 @@
     "@fortawesome/angular-fontawesome": "0.10.1",
     "@fortawesome/fontawesome-svg-core": "1.3.0",
     "@fortawesome/free-solid-svg-icons": "6.0.0",
+    "@ngx-matomo/router": "^2.6.1",
+    "@ngx-matomo/tracker": "^2.6.1",
```
based on the version of angular used the version is ok, once the angular version is updated the matomo client version will be upldated as well.

amydoxym commented on src/app/core/analytics/matomo.directives.ts:
```diff
@@ -0,0 +1,130 @@
+  selector: "[mifosxMatomoClick]",
+})
+export class MatomoClickDirective {
+  @Input() mifosxMatomoClick: string; // Action
```
@gilclefes can we remove the mifosx part and keep only matomoClick

amydoxym commented on src/app/core/analytics/matomo.directives.ts:
```diff
@@ -0,0 +1,130 @@
+  selector: "[mifosxMatomoForm]",
+})
+export class MatomoFormDirective {
+  @Input() mifosxMatomoForm: string; // Form name
```
```suggestion
  @Input() matomoForm: string; // Form name
```

amydoxym commented on src/app/core/analytics/matomo.directives.ts:
```diff
@@ -0,0 +1,130 @@
+ * Usage: <a mifosxMatomoOutbound [href]="externalUrl">External Link</a>
+ */
+@Directive({
+  selector: "[mifosxMatomoOutbound]",
```
idem

amydoxym commented on src/app/core/authentication/authentication.interceptor.ts:
```diff
@@ -102,7 +102,7 @@ export class AuthenticationInterceptor implements HttpInterceptor {
       httpOptions.headers['Platform-TenantId'] = tenantId;
     } else {
-      delete httpOptions.headers['Platform-TenantId'];
+      delete httpOptions.headers["Platform-TenantId"];
```
revert this change

amydoxym commented on src/app/core/core.module.ts:
```diff
@@ -1,40 +1,52 @@
+import { environment } from "environments/environment";
 
 /** Custom Services */
-import { AuthenticationService } from './authentication/authentication.service';
```
please revert the changes of ' to " so we can focus only on the exact logic change

amydoxym commented on src/app/core/shell/sidenav/sidenav.component.html:
```diff
@@ -1,37 +1,49 @@
-  'sidebar-compact': sidenavCollapsed
-  }">
-
+    'sidebar-full': !sidenavCollapsed,
```
what has really changed in here?

gilclefes replied:
move the > sign to the next line which is formatting by the editor

gilclefes committed:
Refactor: Update Matomo tracking attributes and improve code consistency

- Changed `mifosxMatomoClick` to `matomoClick` in view-user.component.html for consistency.
- Updated tracking event strings in view-user.component.ts to use single quotes for consistency.
- Refactored environment configuration files to use single quotes for string values.
- Enhanced TSLint configuration for better code style enforcement.
- Added a script to fix quotes in TypeScript files using TSLint.

simiyu17 reviewed (approved):
Looks good
